### DISCUSSION TITLE
How to get glib/gio and other versions?

### DISCUSSION THREAD

**Discussion by d47081** [gtk-rs/gtk4-rs#1930] [Q&A] [answered]
I'm using [gnome_47](https://github.com/gtk-rs/gtk4-rs/blob/92b10c1b5b957c8db3a3447fe80e3c4720fe1b6d/gtk4/Cargo.toml#L37) preset and want to print it submodules version into about dialog, but this code does not work

```rust
gtk::gio::version()
gtk::glib::version()
```

gtk provides following methods:

```rust
gtk::major_version(),
gtk::minor_version(),
gtk::micro_version()
```
what about gio, glib, pixbuf, cairo and other?

**Comment by sdroege** [comment, marked as answer]
There's no (usable) API for that in GLib/GIO

**Reply by d47081** [reply]
thanks for short answer, 
is any other option to get this info using rust?

**Reply by sdroege** [reply]
Where would you get the numbers from? :)

**Reply by d47081** [reply]
okay, closing if impossible
